elasticsearch-core/src/main/scala/com/sumologic/elasticsearch/restlastic/dsl/QueryDsl.scala

```diff
@@ -368,4 +370,64 @@ trait QueryDsl extends DslCommons with SortDsl {
       _dis_max -> innerMap
     )
   }
+
+  case class HighlightRoot(queryRoot: QueryRoot, highlight: Highlight)
+    extends QueryRootBase {
+
+    override def toJson: Map[String, Any] = {
+      queryRoot.toJson ++ highlight.toJson
+    }
+  }
+
+  case class Highlight(fields: Seq[HighlightField], preTags: Seq[String] = Seq(), postTags: Seq[String] = Seq())
+    extends EsOperation {
+
+    val _pre_tags = "pre_tags"
+    val _post_tags = "post_tags"
+    val _fields = "fields"
+    val _highlight = "highlight"
+
+    val pre_tags = if (preTags.isEmpty) Map[String, Any]() else Map(_pre_tags -> preTags)
+    val post_tags = if (postTags.isEmpty) Map[String, Any]() else Map(_post_tags -> postTags)
+
+    override def toJson: Map[String, Any] = Map(
+      _highlight -> {
+        Map( _fields -> fields.foldLeft(Map[String, Any]())((l, r) => l ++ r.toJson)) ++
```

Comment:
Ah, I should have thought of using reduce instead of foldLeft. It makes the code much cleaner! Thanks for the advice. Updated.